plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -534,7 +534,7 @@ private class DokkaDescriptorVisitor(
                     parameter(index, desc, DRIWithPlatformInfo(dri, actual))
                 },
                 sources = actual,
-                expectPresentInSet = sourceSet.takeIf { isExpect },
+                expectPresentInSet = if(isExpect) sourceSet else null,
```

Comment:
I suppose this can be https://youtrack.jetbrains.com/issue/KT-52561 that is fixed in https://github.com/JetBrains/kotlin/commit/c526ed7300c5d8ee838185f6cb8a6a1c63092504 (Kotlin 1.7.0)